File: apollo-runtime/src/main/java/com/apollographql/apollo/Logger.java
package com.apollographql.apollo;

import com.apollographql.apollo.api.internal.Optional;

import javax.annotation.Nonnull;

public interface Logger {
  int DEBUG = 3;
  int WARN = 5;
  int ERROR = 6;

  void log(int priority, @Nonnull String message, @Nonnull Optional<Throwable> t, @Nonnull Object... args);
}

File: apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/Logger.kt
package com.apollographql.apollo3

/**
 * Logger to use for logging by the [com.apollographql.apollo3.ApolloClient]
 */
interface Logger {
  /**
   * Logs the message to the appropriate channel (file, console, etc)
   *
   * @param priority the priority to set
   * @param message message to log
   * @param t Optional throwable to log
   * @param args extra arguments to pass to the logged message.
   */
  fun log(priority: Int, message: String, t: Throwable?, vararg args: Any)

  companion object {
    const val DEBUG = 3
    const val WARN = 5
    const val ERROR = 6
  }
}
